contracts/02-tamagotchi-interaction/src/lib.rs

```diff
@@ -1,18 +1,64 @@
 #![no_std]
 
 #[allow(unused_imports)]
-use gstd::prelude::*;
+use gstd::{exec, msg, prelude::*, ActorId};
+use tamagotchi_interaction_io::*;
+
+#[derive(Default, Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
+pub struct Tamagotchi {
+    pub name: String,
+    pub date_of_birth: u64,
+    pub owner: ActorId,
+    pub fed: u64,
+    pub fed_block: u64,
+    pub entertained: u64,
+    pub entertained_block: u64,
+    pub slept: u64,
+    pub slept_block: u64,
+}
+
+impl Tamagotchi {
+    fn current_fed(&mut self) -> u64 {
+        let a: u64 =
+            self.fed - (HUNGER_PER_BLOCK as u64) * ((exec::block_height() as u64) - self.fed_block);
+        a
+    }
+    fn current_entertained(&mut self) -> u64 {
+        let b: u64 = self.entertained
+            - (BOREDOM_PER_BLOCK as u64) * ((exec::block_height() as u64) - self.entertained_block);
```

Comment:
Using `self.entertained_block` is not correct here. We need to save the last processed block height here.
```suggestion
            - (BOREDOM_PER_BLOCK as u64) * ((exec::block_height() as u64) - last_processed_block);
```

Reply:
I have fixed it.